frontend/lib/framework/parser-error-classes-list.cpp

```diff
@@ -96,6 +96,97 @@ void ErrorCommentEOF::write(ErrorWriterBase& wr) const {
+      labelName = target->name().str();
+    }
+  } else if (auto cont = invalidAst->toContinue()) {
+    astType = "continue";
+    if (auto target = cont->target()) {
+      labelName = target->name().str();
+    }
+  } else {
+    CHPL_ASSERT(false && "not a control flow element handled by this error");
+  }
+
+  std::string blockingName = "";
+  if (!blockingAst) {
+    // Nothing
+  } else if (blockingAst->isForall()) {
+    blockingName = "a 'forall' statement";
+  } else if (blockingAst->isCoforall()) {
+    blockingName = "a 'coforall' statement";
+  } else if (blockingAst->isOn()) {
+    blockingName = "an 'on' statement";
+  } else if (blockingAst->isBegin()) {
+    blockingName = "a 'begin' statement";
+  } else if (blockingAst->isSync()) {
+    blockingName = "a 'sync' statement";
+  } else if (blockingAst->isCobegin()) {
+    blockingName = "a 'cobegin' statement";
+  } else if (auto fn = blockingAst->toFunction()) {
+    if (fn->kind() == uast::Function::Kind::ITER) {
+      blockingName = "an iterator";
+    } else {
+      blockingName = "a procedure";
+    }
+  } else {
+    CHPL_ASSERT(false && "not a blocking element handled by this error");
+  }
+
+
+  if (!labelName.empty()) {
+    // labeled break or continue did not find the target thing.
+    wr.heading(kind_, type_, invalidAst, "could not find label '", labelName,
+               "' for '", astType, "' statement.");
+    wr.code(invalidAst, { invalidAst });
+
+    if (blockingAst) {
+      // Didn't go all the way to the top, but stopped at a particular loop or fn.
+      wr.note(blockingAst, "stopped searching here because '", astType, "' "
+              "statements are not allowed to jump outside ", blockingName, ":");
+      wr.code(blockingAst, { blockingAst });
+    }
+  } else if (!blockingAst || blockingAst->isFunction()) {
+    // break or continue without a label, outside any loop.
+    std::string missingThing = "";
+    if (invalidAst->isReturn()) {
+      missingThing = "a procedure or an iterator";
+    } else if (invalidAst->isYield()) {
+      missingThing = "an iterator";
+    } else {
+      missingThing = "a loop";
+    }
+    wr.heading(kind_, type_, invalidAst, "'", astType,
+               "' is not allowed outside of ", missingThing, ".");
+    wr.code(invalidAst, { invalidAst });
+    if (blockingAst) {
+      CHPL_ASSERT(invalidAst->isBreak() || invalidAst->isContinue());
+      wr.note(blockingAst, "stopped searching here because '", astType, "' "
+              "statements are not allowed to jump outside ", blockingName, ":");
+      wr.code(blockingAst, { blockingAst });
+    }
+  } else {
+    // any piece of control flow banned within a particular language feature
+    wr.heading(kind_, type_, invalidAst, "'", astType, "' is not allowed in ",
+               blockingName, ".");
+    wr.code(invalidAst, { invalidAst });
+    // wr.message("The mentioned '", blockingName, "' statement is here:");
+    // wr.code(blockingAst, { blockingAst });
```

Comment:
did you mean to leave this commented-out code in here? Should it turn into a TODO?